Training/tf/create_dataset.py

```diff
@@ -0,0 +1,126 @@
+                if sum_:=np.sum(a[tau_type_column]!=a["tauType"]):
+                    print(f'\n        [WARNING] non-zero fraction of recomputed tau types: {sum_/len(a["tauType"])*100:.1f}%\n')
+            else:
+                tau_type_column = 'tauType'
+
+            # create one-hot encoded labels
+            label_columns = []
+            for tau_type, tau_type_value in tau_type_map.items():
+                a[f'label_{tau_type}'] = ak.values_astype(a[tau_type_column] == tau_type_value, np.int32)
+                label_columns.append(f'label_{tau_type}')
+            with open_dict(cfg):
+                cfg["label_columns"] = label_columns
+  
+            time_2 = time.time()
+            print(f'        Preprocessing: took {(time_2-time_1):.1f} s.')
+
+            for tau_type in tau_types:
+                time_2 = time.time()
+                # select only given tau_type
+                a_selected = a[a[f'label_{tau_type}'] == 1]
+                n_selected = len(a_selected)
+                n_samples[tau_type] += n_selected
+                print(f'        Selected: {tau_type}: {n_selected} samples')
+
+                # final tuple with elements to be stored into TF dataset
+                data = ()
+
+                # add awkward arrays converted to TF ragged arrays
+                for particle_type, feature_names in cfg['feature_names'].items(): # do this separately for each particle collection
+                    X = awkward_to_tf(a_selected, particle_type, feature_names) # will keep only feats from feature_names
+                    data += (X,)
+                
+                # all labels to final dataset
+                y = ak.to_pandas(a_selected[cfg["label_columns"]]).values
+                data += (y,)
+
+                # add additional columns if needed
+                if dataset_cfg['add_columns'] is not None:
+                    add_columns = ak.to_pandas(a_selected[dataset_cfg['add_columns']])
+                    add_columns = np.squeeze(add_columns.values)
+                    data += (add_columns,)
+                
+                # create TF dataset 
+                dataset = tf.data.Dataset.from_tensor_slices(data)
+                time_3 = time.time()
+                print(f'        Preparing TF datasets: took {(time_3-time_2):.1f} s.')
+
+                # remove existing datasets
+                path_to_dataset = to_absolute_path(f'{cfg["path_to_dataset"]}/{cfg["dataset_name"]}/{dataset_type}/{os.path.splitext(os.path.basename(file_name))[0]}/{tau_type}')
+                if os.path.exists(path_to_dataset):
+                    shutil.rmtree(path_to_dataset)
+                else:
+                    os.makedirs(path_to_dataset, exist_ok=True)
+
+                # save
+                tf.data.experimental.save(dataset, path_to_dataset)
```

Comment:
According to the new release docs, they now moved it from experimental to the tf.data class methods: https://www.tensorflow.org/api_docs/python/tf/data/Dataset#save